components/dashboard/src/start/StartWorkspace.tsx

```diff
@@ -366,51 +372,56 @@ export default class StartWorkspace extends React.Component<StartWorkspaceProps,
     }
 
     async ensureWorkspaceAuth(instanceID: string, retry: boolean) {
-        if (!document.cookie.includes(`${instanceID}_owner_`)) {
-            let attempt = 0;
-            let fetchError: Error | undefined = undefined;
-            while (attempt <= 10) {
-                attempt++;
-
-                let code: number | undefined = undefined;
-                fetchError = undefined;
-                try {
-                    const authURL = gitpodHostUrl.asWorkspaceAuth(instanceID);
-                    const response = await fetch(authURL.toString());
-                    code = response.status;
-                } catch (err) {
-                    fetchError = err;
-                }
+        if (document.cookie.includes(`${instanceID}_owner_`)) {
+            // Cookie already present
+            return;
+        }
 
-                if (retry && (code === 404 || !!fetchError)) {
-                    console.warn("Unable to retrieve workspace-auth cookie! Retrying shortly...", fetchError, {
-                        code,
-                        attempt,
-                    });
-                    // If the token is not there, we assume it will appear, soon: Retry a couple of times.
-                    await new Promise((resolve) => setTimeout(resolve, 2000));
-                    continue;
-                }
-                if (code !== 200) {
-                    // getting workspace auth didn't work as planned
-                    console.error("Unable to retrieve workspace-auth cookie! Quitting.", {
-                        code,
-                        attempt,
-                    });
-                    return;
-                }
+        const MAX_ATTEMPTS = 10;
+        const ATTEMPT_INTERVAL = 2000; // [ms]
+        let attempt = 0;
+        let fetchError: Error | undefined = undefined;
+        while (attempt <= MAX_ATTEMPTS) {
+            attempt++;
 
-                // Response code is 200 at this point: done!
-                console.info("Retrieved workspace-auth cookie.", { code, attempt });
+            let code: number | undefined = undefined;
+            fetchError = undefined;
+            try {
+                const authURL = gitpodHostUrl.asWorkspaceAuth(instanceID);
+                const response = await fetch(authURL.toString());
+                code = response.status;
+            } catch (err) {
+                fetchError = err;
+            }
+
+            if (retry && (code === 404 || !!fetchError)) {
+                console.warn("Unable to retrieve workspace-auth cookie! Retrying shortly...", fetchError, {
+                    code,
+                    attempt,
+                });
+                // If the token is not there, we assume it will appear, soon: Retry a couple of times.
+                await new Promise((resolve) => setTimeout(resolve, ATTEMPT_INTERVAL));
+                continue;
+            }
```

Comment:
What error does the user get when we always execute this if in the retry up to MAX_ATTEMPTS? The `fetchError` would be undefined, so we wouldn't throw anything when we exit this method.

Reply:
Once we hit MAX_ATTEMPTS, the `continue` leads us to exit the loop with `fetchError` set, and re-trhow down [here](https://github.com/gitpod-io/gitpod/blob/e712bdf067ae33ea69d3189f592f1ac70bc25f57/components/dashboard/src/start/StartWorkspace.tsx#L425-L428).

Reply:
Where is the `fetchError` set?

```
loop 1:
fetch doesn't throw, returns 404
if gets executed {
  continue..
}

the same happens on subsequent loops

while condition does not hold, exit loop

fetchError is not set
```

Reply:
Ah. That behavior was maintained from prior code. But after testing all call-sites, we really can make this an `ensureWorkspaceAuth`.

Mapped 404 to new Error :heavy_check_mark: